### DISCUSSION TITLE
A better way of plotting time?

### DISCUSSION THREAD

**Discussion by mcrookrumsey** [esoreq/dcarte#8] [Q&A]
I often find myself wanting to plotting the count of _events_ (e.g., bed2bed transitions) over time. While I'm able to produce the plot I need I'm certain there is a better way of doing it. Below is an example of code I use to get the distribution of events through the night. 

```
import datetime as dt
from matplotlib.dates import date2num
from datetime import datetime

df['time'] = df['start_date'].dt.time
df['time'] = df['time'].astype(str)
df['time'] = pd.to_datetime(df['time'])

import matplotlib as mpl
def myFormatter(x, pos):
       return pd.to_datetime(x).strftime('%H:%M')
ax.xaxis.set_major_formatter(mpl.ticker.FuncFormatter(myFormatter))

fig, ax = plt.subplots(figsize=(10,8))
hallway_plot = sns.histplot(BTm_hallway_hour_count["time"].astype(np.int64), bins=36) # each bin is 30mins
ax.xaxis.set_major_formatter(mpl.ticker.FuncFormatter(myFormatter))
```
![image](https://user-images.githubusercontent.com/83961548/161076501-0ab71847-34e8-43a5-97e1-95f8f353a012.png)

As you can see, it's not great. I can get the info I want but it is not easiest plot to interpret. Is there a better way to write this? Would it be better to plot midday to the following midday?

**Comment by esoreq** [comment]
So there are many ways of doing this (that depend on the visualisation package) I suggest you add this to the [dcarte_tutorials discussion](https://github.com/esoreq/dcarte_tutorials/discussions) and I will prioritise a session on this topic
